### ISSUE TITLE
[Bug] PyGRASS: Confusing TypeError on VectorTopo.open.

### ISSUE THREAD

**Issue by Viech** [OSGeo/grass#1555]
**Describe the bug**
I can read `archsites` of `spearfish_grass70data_0` with `VectorTopo` just fine but when I use `v.select` to filter points not in the computational region, opening the resulting vector map with `VectorTopo` raises `TypeError: Driver is not supported yet, pleas use: sqlite or pg`. I suspect that the filtered map is missing some important metadata (I'm happy to receive a hint on how to make it work) but in any case **the raised `TypeError` is confusing**, as I did not specify any database driver. Additional error handling seems necessary to tell the user exactly why the map can't be loaded.

**To Reproduce**
```python
#!/usr/bin/env python3

from grass.pygrass.modules import Module
from grass.pygrass.vector import VectorTopo

try:
    a = VectorTopo("archsites", mode="r")
    a.open()  # works

    Module("v.in.region", output="test_region", quiet=True)
    Module(
        "v.select",
        ainput="archsites",
        binput="test_region",
        operator="overlap",
        output="test_filtered",
        quiet=True,
    )

    b = VectorTopo("test_filtered", mode="r", quiet=True)
    b.open()  # confusing error
finally:
    Module(
        "g.remove",
        type="vector",
        name=("test_region", "test_filtered"),
        flags="f",
        quiet=True,
    )
```
Output:
```
Traceback (most recent call last):
  File "…/test.py", line 21, in <module>
    b.open()  # misleading error
  File "/opt/grass/etc/python/grass/pygrass/vector/abstract.py", line 393, in open
    self.table = self.dblinks.by_layer(layer).table()
  File "/opt/grass/etc/python/grass/pygrass/vector/table.py", line 792, in table
    return Table(self.table_name, self.connection(), self.key)
  File "/opt/grass/etc/python/grass/pygrass/vector/table.py", line 775, in connection
    raise TypeError(str_err)
TypeError: Driver is not supported yet, pleas use: sqlite or pg
```

**Expected behavior**
Opening the map should either work, if possible, or raise an exception that makes it clear why it would not work.

**Screenshots**
–

**System description (please complete the following information):**

- Operating System: Linux 5.11.16-arch1-1
- GRASS GIS version: 7.8.5

```shell
% g.version -rge
version=7.8.5
date=2020
revision=4e3f393
build_date=2021-04-27
build_platform=x86_64-pc-linux-gnu
build_off_t_size=8
libgis_revision=2021-04-27T07:33:34+00:00
libgis_date=2021-04-27T10:00:00+02:00
proj=8.0.0
gdal=3.2.2
geos=3.9.1
sqlite=3.35.5
```
```shell
% python3 -c "import sys, wx; print(sys.version); print(wx.version())"
3.9.4 (default, Apr 20 2021, 15:51:38) 
[GCC 10.2.0]
4.1.1 gtk3 (phoenix) wxWidgets 3.1.5
```

**Additional context**
–

**Comment by Viech** [comment]
I **think** running `db.connect` without arguments once fixed the issue. Previously `v.db.connect --help` mentioned `default: dbf` for the `driver` option, now it says `default: sqlite`. So maybe there is some problem on my platform with how that default value is determined initially?

I still think the error message should be more precise: Which unsupported driver is selected and how was it selected (e.g. was it selected because the map has a connection to such a database or because this driver is set as a default somewhere)?

**Comment by ninsbl** [comment]
Hi @Viech , the `spearfish_grass70data_0` data is a quite old sample dataset. I assume the `archsites` vector map is connected to a dbf database backend. That databse connection (and the driver) were defined when the data was created. Thus the error message that the driver of the map is not supported.

If you want to update the data, please hae a look at:
https://grass.osgeo.org/grass78/manuals/v.db.reconnect.all.html

**Comment by Viech** [comment]
Hey @ninsbl,

note that I could load `archsites` itself just fine, it was only when I created a new vector map (using `v.select` on `archsites`) that a `dbf` database was seemingly associated with the new map. It seems that `dbf` was either a default driver for new maps for (1) just the old spearfish location or (2) my grass environment or (3) some other scope I do not know about. Indeed, `v.db.connect --help` read `default: dbf` at some point. It looks like I changed that default to `sqlite` by just running `db.connect` with no arguments.

**Comment by lucadelu** [comment]
yes it is normal that you can work fine with normal GRASS interface. pyGRASS is low level library and it use python library to manage db connect. at the time only sqlite and postgresql backend are supported. 

Any improvement is really welcome.

**Comment by Viech** [comment]
Hi @lucadelu,

I could load `archsites` fine **within Python**. With the same Python code, loading a map that was directly derived from `archsites` via `v.select` failed. Please read "To Reproduce" in my initial post for clarity.

**TL;DR for this issue**: I originally assumed that an error message I got was misleading or wrong as I did not ever configure a nonstandard database driver at any point yet the error message was related to nonstandard database drivers. I realize now that the error message was technically correct and that the problem was elsewhere: Somehow my default database driver (for some scope) was set to `dbf` instead of `sqlite`. I still believe that the error message is not detailed enough for an average user to understand what exactly went wrong. The error should state which problematic driver is requested and what requests it (probably the map).

**Comment by ninsbl** [comment]
@Viech : This is probably not a very common issue, but I agree that the error message can be confusing.

The relevant code is here:
https://github.com/OSGeo/grass/blob/632116c55805f2e1b30517cbf263497bfadc5913/python/grass/pygrass/vector/table.py#L716-L720

Could you come up with a suggestion for an improved error message? In particular, what kind of error type would you expect.
Would be great if you clould create a PR.

**Comment by Viech** [comment]
I'll have a look at this later. Can you assign me to this issue?

(The error should probably be a `NotImplementedError` and `str_err` should mention `driver`. Then, one could catch that exception in `Info.open` and raise an even more descriptive error `from` the original one along the lines of `Failed to load the existing attribute table for layer <layer> of <map>.`)

**Comment by ninsbl** [comment]
> I'll have a look at this later. Can you assign me to this issue?

Great, thanks!
